lib/streamlit/elements/markdown.py

```diff
@@ -96,10 +96,14 @@ def header(self, body):
 
         """
         header_proto = MarkdownProto()
-        header_proto.body = "## %s" % clean_text(body)
+        if anchor is None:
+            header_proto.body = '## %s' % clean_text(body)
+        else:
+            header_proto.body = '<h2 data-anchor="%s">%s</h2>' % (anchor, clean_text(body))
```

Comment:
I believe anchors are generally populated as the id or name attribute. Multipage app is something we will want to do later and we may need to switch to id later  depending how we proceed.

Reply:
Addressing this comment and your [other comment](https://github.com/streamlit/streamlit/pull/2513#discussion_r547590060) together:

Here all we're doing is finding a way to pass the anchor to the frontend, which eventually intercepts the `data-anchor` in the ReactMarkdown renderer, and never actually prints it out in the `<h2>` element.

The `<a data-anchor="...">` is because I used `name` at first, but apparently React doesn't support the `name` property on `<a>` tags. Don't know why.

Can you say more about multipage app and how that relates to using id?

Reply:
Update: I just removed the `<a data-anchor>` altogether as it wasn't doing anything.

Reply:
> Can you say more about multipage app and how that relates to using id?

My first thought for multipage app is using react router and I believe the hash link from react router would scroll to the element matching the ID. Then if we passed the ID here we could just spit out the html as is without doing any modifications.